table/block.cc

```diff
@@ -63,6 +63,35 @@ struct DecodeEntry {
   }
 };
 
+struct DecodeEntrySlow {
+  inline const char* operator()(const char* p, const char* limit,
+                                uint32_t* shared, uint32_t* non_shared,
+                                uint32_t* value_length) {
+    // We need 2 bytes for shared and non_shared size. We also need one more
+    // byte either for value size or the actual value in case of value delta
+    // encoding.
+    assert(limit - p >= 3);
```

Comment:
Rather than assert, avoid reading data out of bound?

Reply:
Done.